crankshaft-engine/src/service/runner/backend/generic.rs

```diff
@@ -123,7 +128,7 @@ impl crate::Backend for Backend {
                 })
                 .transpose()?;
 
-            for execution in task.executions {
+            for execution in task.executions.clone() {
```

Comment:
I think we could perhaps change this to `std::mem::take(&mut task.executions)` rather than cloning it.

I assume this was necessary because we're borrowing `task` below for the new call to `send_event`?

Reply:
I forgot that `executions` was a `NonEmpty` and not a `Vec`, so we can't do what I suggested.

As we're already calling `to_string` below on the task name, we can just move it to above the loop and remove the clone here.